syslog: T4251: Add documentation for syslog TLS configuration (backport)

alexandr-san4ez opened the pull request:
## Change Summary
Documented protocol, certificate, auth-mode, and permitted-peers requirements for TLS configuration.
https://vyos--1689.org.readthedocs.build/en/1689/configuration/system/syslog.html

## Related Task(s)
<!-- optional: Link related Tasks on Phabricator. -->
* https://vyos.dev/T4251

## Related PR(s)
* https://github.com/VyOS-Networks/vyos-1x/pull/1246
* https://github.com/vyos/vyos-1x/pull/4734

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-documentation/blob/current/CONTRIBUTING.md) document

alexandr-san4ez committed:
syslog: T4251: Add documentation for syslog TLS configuration

Documented protocol, certificate, auth-mode, and permitted-peers
requirements for TLS configuration.

(cherry picked from commit a3d3347ee8e05e11a23d795070803748bcf84458)

dmbaturin commented:
@alexandr-san4ez I'm converting it to draft until it's merged into LTS branches.

alexandr-san4ez committed:
syslog: T4251: Fix documentation of TLS enablement logic for syslog

(cherry picked from commit a271d322e2977686dbd67f5967a9c1660f6789bd)

alexandr-san4ez committed:
syslog: T4251: Rename "permitted-peers" to "permitted-peer"

- Renamed `permitted-peers` to `permitted-peer`.
- Added support for multiple `permitted-peer` entries.

(cherry picked from commit 85a097b4a19b52179b45966cbcc8c7f30030a884)
